File: Backend/src/main/java/unical/demacs/rdm/persistence/entities/Job.java
package unical.demacs.rdm.persistence.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import unical.demacs.rdm.persistence.enums.JobPriority;
import unical.demacs.rdm.persistence.enums.JobStatus;

@Entity
@Table(name = "jobs")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder(builderMethodName = "buildJob")
public class Job {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column
    private String description;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private JobStatus status = JobStatus.PENDING;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User assignee;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private JobPriority priority = JobPriority.LOW;

    @Column(nullable = false)
    private long duration;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "machine_type_id", nullable = false)
    private MachineType requiredMachineType;
}

File: Frontend/src/app/generated-api/model/job.ts

import { MachineType } from './machineType';
import { User } from './user';

export interface Job {
    id?: number;
    title?: string;
    description?: string;
    status?: Job.StatusEnum;
    assignee?: User;
    priority?: Job.PriorityEnum;
    duration?: number;
    requiredMachineType?: MachineType;
}
export namespace Job {
    export type StatusEnum = 'PENDING' | 'SCHEDULED' | 'IN_PROGRESS' | 'COMPLETED' | 'CANCELLED';
    export const StatusEnum = {
        PENDING: 'PENDING' as StatusEnum,
        SCHEDULED: 'SCHEDULED' as StatusEnum,
        INPROGRESS: 'IN_PROGRESS' as StatusEnum,
        COMPLETED: 'COMPLETED' as StatusEnum,
        CANCELLED: 'CANCELLED' as StatusEnum
    };
    export type PriorityEnum = 'LOW' | 'MEDIUM' | 'HIGH' | 'URGENT';
    export const PriorityEnum = {
        LOW: 'LOW' as PriorityEnum,
        MEDIUM: 'MEDIUM' as PriorityEnum,
        HIGH: 'HIGH' as PriorityEnum,
        URGENT: 'URGENT' as PriorityEnum
    };
}
